core/admin/mailu/sso/views/base.py

```diff
@@ -57,3 +59,36 @@ def logout():
     flask.session.destroy()
     return flask.redirect(flask.url_for('.login'))
 
+
+@sso.route('/proxy', methods=['GET'])
+@sso.route('/proxy/<target>', methods=['GET'])
+def proxy(target='webmail'):
+    ip = ipaddress.ip_address(flask.request.remote_addr)
+    if not any(ip in cidr for cidr in app.config['PROXY_AUTH_WHITELIST']):
+        return flask.abort(500, '%s is not on PROXY_AUTH_WHITELIST' % flask.request.remote_addr)
+
+    email = flask.request.headers.get(app.config['PROXY_AUTH_HEADER'], None)
+    if not email:
+        return flask.abort(500, 'No %s header' % app.config['PROXY_AUTH_HEADER'])
+
+    user = models.User.get(email)
+    if user:
+        flask.session.regenerate()
+        flask_login.login_user(user)
+        return flask.redirect(app.config['WEB_ADMIN'] if target=='admin' else app.config['WEB_WEBMAIL'])
+    elif app.config['PROXY_AUTH_CREATE']:
+        localpart, desireddomain = email.split('@')
+        domain = models.Domain.query.get(desireddomain) or flask.abort(500, 'You don\'t exist. Go away! (domain=%s)' % desireddomain)
```

Comment:
I think the error response should be the same as for the exception:
"Invalid email address".
Everything else (domain not found) should only be logged.

Reply:
Usually I'd agree that it's important to leak as little information as possible... but here this is users who are already authenticated (not on Mailu but by the proxy) and it will help helpdesk to have a more descriptive message.

Any particular reason why you want this changed?

Reply:
It only was about not leaking too much information.
Maybe the we should better abort with "Domain does not exist", shouldn't we?
But feel free to resolve this conversation.

Reply:
But now there's another error. `flask.abort` raises an exception.
If you wrap the whole block after line 85 in a try/except - this HTTPError will be catched.

Reply:
I've reduced the scope in 12117cef